src/auth/credential-provider.ts

```diff
@@ -1,29 +1,47 @@
 import {decodeJwt} from '../utils/jwt';
 
-export interface ICredentialProvider {
+/**
+ * Provides information that the SimpleCacheClient needs in order to establish a connection to and authenticate with
+ * the Momento service.
+ * @export
+ * @interface CredentialProvider
+ */
+export interface CredentialProvider {
+  /**
+   * @returns {string} Auth token provided by user, required to authenticate with the service
+   */
   getAuthToken(): string;
 
+  /**
+   * @returns {string} The host which the Momento client will connect to for Momento control plane operations
+   */
   getControlEndpoint(): string;
 
+  /**
+   * @returns {string} The host which the Momento client will connect to for Momento data plane operations
+   */
   getCacheEndpoint(): string;
-
-  getTrustedControlEndpointCertificateName(): string | null;
-
-  getTrustedCacheEndpointCertificateName(): string | null;
 }
 
-export class EnvMomentoTokenProvider implements ICredentialProvider {
+/**
+ * Reads and parses a momento auth token stored as an environment variable.
+ * @export
+ * @class EnvMomentoTokenProvider
+ */
+export class EnvMomentoTokenProvider implements CredentialProvider {
   private readonly authToken: string;
   private readonly controlEndpoint: string;
   private readonly cacheEndpoint: string;
-  private readonly trustedControlEndpointCertificateName: string | null;
-  private readonly trustedCacheEndpointCertificateName: string | null;
+
+  /**
+   * @param {string} envVariableName the name of the environment variable from which the auth token will be read
+   * @param {string | undefined} controlEndpoint optionally overrides the default controlEndpoint
+   * @param {string | undefined} cacheEndpoint optionally overrides the default cacheEndpoint
```

Comment:
[According to TypeScript](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#param-and-returns) the way to make a param optional is to enclose the name in square brackets.

```suggestion
   * @param {string} [controlEndpoint] optionally overrides the default controlEndpoint
   * @param {string} [cacheEndpoint] optionally overrides the default cacheEndpoint
```